### PR TITLE
[Enhancement] Display inactive reason for materialized view

### LINKED ISSUE #21501: Display inactive reason of materialized view
## Enhancement

image

Once a materialized view is inactive, no method could be used to diagnose it. So we must have to provide some diagnose method for it

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #21501

## Problem Summary(Required) ：

```
***************************[ 2. row ]***************************
MATERIALIZED_VIEW_ID                 | 90079
TABLE_SCHEMA                         | hehe
TABLE_NAME                           | mv_foo
REFRESH_TYPE                         | INCREMENTAL
IS_ACTIVE                            | false
INACTIVE_REASON                      | base table dropped: 10087
PARTITION_TYPE                       | UNPARTITIONED
TASK_ID                              |
TASK_NAME                            |
LAST_REFRESH_START_TIME              |
LAST_REFRESH_FINISHED_TIME           |
LAST_REFRESH_DURATION                |
LAST_REFRESH_STATE                   |
LAST_REFRESH_FORCE_REFRESH           |
LAST_REFRESH_START_PARTITION         |
LAST_REFRESH_END_PARTITION           |
LAST_REFRESH_BASE_REFRESH_PARTITIONS |
LAST_REFRESH_MV_REFRESH_PARTITIONS   |
LAST_REFRESH_ERROR_CODE              |
LAST_REFRESH_ERROR_MESSAGE           |
TABLE_ROWS                           | CREATE MATERIALIZED VIEW `mv_foo`
COMMENT "MATERIALIZED_VIEW"
DISTRIBUTED BY HASH(`id`) BUCKETS 2
REFRESH INCREMENTAL
PROPERTIES (
"replication_num" = "1",
"storage_medium" = "HDD"
)
AS SELECT `hehe`.`id`
FROM `hehe`.`hehe`;
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by LiShuMing — DISMISSED** [review]
Good Work~

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -3177,9 +3177,7 @@ private void disableMaterializedViewForRenameTable(Database db, OlapTable olapTa
         for (MvId mvId : olapTable.getRelatedMaterializedViews()) {
             MaterializedView mv = (MaterializedView) db.getTable(mvId.getId());
             if (mv != null) {
-                LOG.warn("Setting the materialized view {}({}) to invalid because " +
-                        "the table {} was renamed.", mv.getName(), mv.getId(), olapTable.getName());
-                mv.setActive(false);
+                mv.setInactiveAndReason("base table renamed: " + olapTable.getName());
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3180]: better to keep the log

Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3256]: done

**Review by Astralidea — DISMISSED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Review by Astralidea — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 37 (43.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|16|00.00%|[506, 507, 509, 510, 511, 512, 513, 514, 515, 516, 517, 518, 519, 521, 522, 524]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|[1487, 1495]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|[567]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|7|71.43%|[675, 1071]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/MaterializedViewsSystemTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowMaterializedViewsStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|3|3|100.00%|[]|

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 37 (43.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|16|00.00%|[506, 507, 509, 510, 511, 512, 513, 514, 515, 516, 517, 518, 519, 521, 522, 524]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|[1567, 1575]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|[567]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|7|71.43%|[675, 1071]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/MaterializedViewsSystemTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowMaterializedViewsStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 37 (43.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|16|00.00%|[506, 507, 509, 510, 511, 512, 513, 514, 515, 516, 517, 518, 519, 521, 522, 524]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|2|00.00%|[1567, 1575]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|1|2|50.00%|[567]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|7|71.43%|[675, 1071]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/system/information/MaterializedViewsSystemTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/LakeTableSchemaChangeJob.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowMaterializedViewsStmt.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeJobV2.java|3|3|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

File: be/src/exec/schema_scanner/schema_materialized_views_scanner.cpp [hunk 2]
```diff
@@ -132,6 +133,14 @@ Status SchemaMaterializedViewsScanner::fill_chunk(ChunkPtr* chunk) {
...
         }
         case 6: {
+            // inactive_reason
+            ColumnPtr column = (*chunk)->get_column_by_slot_id(slot_id);
+            const std::string* str = &tbl_status.inactive_reason;
+            Slice value(str->c_str(), str->length());
+            fill_column_with_slot<TYPE_VARCHAR>(column.get(), (void*)&value);
+            break;
+        }
+        case 7: {
```
Comment (Seaven) [inline be/src/exec/schema_scanner/schema_materialized_views_scanner.cpp:143]: Will there be compatibility questions with the BE upgrade?

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**